fineftp-server/src/ftp_session.cpp

```diff
@@ -1304,7 +1288,7 @@ namespace fineftp
   // FTP data-socket receive
   ////////////////////////////////////////////////////////
 
-  void FtpSession::receiveFile(std::shared_ptr<IoFile> file)
+  void FtpSession::receiveFile(const std::shared_ptr<IoFile>& file)
```

Comment:
I am not even sure for `shared_ptr` if pass by value is better? But probably not ;)

Reply:
No, absolutely not. A shared_ptr is not a primitive object and copying it not only involves copying the data, but also increasing (and later decreasing) the reference counter.